### ISSUE TITLE
[import/extensions] complains on importing a relative folder

### ISSUE THREAD

**Issue by codehimanshu** [import-js/eslint-plugin-import#2141]
The import statement I used in `module.js` looks like this:

> const { errors } = require('../config')

My directory structure is like this:
```
app
├── src
│   ├── module.js
├── config
│   ├── index.js
│   ├── errors.js
|...
```

Since this is a relative import and the module is a folder which exports all its submodules from `index.js`, `import/extensions` should not complaint about this.

**Comment by ljharb** [comment]
How do you have the extensions rule configured?

**Comment by ljharb** [comment]
@maicol07 ok, and what filename are you importing, with what specifier?

**Comment by maicol07** [comment]
@ljharb a folder, in my case it's a subpackage of [`mithril`](https://mithril.js.org):
```
import stream from 'mithril/stream`
```

`mithril/stream` is a folder

**Comment by ljharb** [comment]
Sure, but it maps to a JS file, so the rule requires the extension, per your config. https://unpkg.com/browse/mithril@2.0.4/stream/stream.js

Although, it has its own package.json, which suggests it's published as `mithril-stream` - is it also available bundled?

**Comment by maicol07** [comment]
Yes, but they will probably remove it in future versions. What about adding a setting like `ignorePackages` for this one?

**Comment by ljharb** [comment]
Why do you want to force `.js` for your first-party files? How does that rationale not apply to files from packages?

**Comment by maicol07** [comment]
No, I don't mean I want to use `ignorePackages`. I want a new option to ignore files with a folder with the same name

**Comment by ljharb** [comment]
ok, so whether it's relative or in a package, you want any `foo/bar/bar.js` to be `foo/bar`? that's just a coincidence here tho - the default "main" for a folder is `index.js`, and it's arbitrary that mithril has named the main file "stream.js".

What I'm asking is, why do you have js set to "always", instead of the much more common "never"? (The answer to that question is important, because it informs what would make sense as a rule option here)

**Comment by maicol07** [comment]
Because I have other packages, such as `lit` where you must import directives by file.
Example:
```
import {
  customElement,
  property
} from 'lit/decorators.js';
import {ifDefined} from 'lit/directives/if-defined.js';
```

**Comment by ljharb** [comment]
Sure - but if `mithril/stream` is ok, why isn't `lit/decorators` ok? If `lit/decorators.js` is preferred, why isn't `mithril/stream/stream.js` preferred?

To me, they're all in the same bucket - either neither of them have extensions (the ecosystem's overwhelming preference), or both of them do.

**Comment by maicol07** [comment]
That's because I would like to use `mithril/stream`, since it is shorter.
The lit one is preferred because it cannot be imported in any other way.
So, while I can write `mithril/stream`, ai can't write `lit/decorators`

**Comment by ljharb** [comment]
Why can't it be imported another way? `lit/decorators` should work just fine, since the `.js` is implicit in transpiled ESM and CJS.

**Comment by ljharb** [comment]
I do see line 26 in https://unpkg.com/browse/lit@2.2.0/package.json - so this seems more like you should want eslint to be configured for "never", but `lit` has used `exports` to force you to write it with the bloat of the extension.

In other words, it seems like _lit_ needs the configuration exception here (until we support "exports", at which point it'd automatically work) - but better would be to ask lit to add an exports entry that omits the extension.

**Comment by maicol07** [comment]
It seems lit Devs won't change that: https://github.com/lit/lit/discussions/2626#discussioncomment-2313720

**Comment by soryy708** [comment]
Looks like the issue is either with a different package or with OPs configuration.
